draft/0004-experience-api.md

```diff
@@ -0,0 +1,923 @@
+- **Social identity**: An authenticated social identity used to directly identify and verify a user.
+- **SSO identity**: An authenticated SSO identity used to directly identify and verify a user.
+
+#### 4.3.2 Multi-factor authentication (MFA) verification
+
+MFA is a security process that requires more than one verification factor to confirm a user's identity. It can be used as an additional security check during user interaction.
+
+Developers can enable MFA using Logto's sign-in experience settings. Once enabled, users will be prompted to provide additional MFA verification records during the interaction. MFA verifications can only be created and verified after the user is identified.
+
+The following are the MFA verification record types supported by Logto:
+
+- **TOTP**: A time-based one-time code generated by a TOTP app for identity verification.
+- **WebAuthn**: A webAuthn (passkey) credential used to verify a user's identity.
+- **Backup code**: A set of backup codes used to verify a user's identity when other MFA verification methods are unavailable.
+
+#### 4.3.3 New profile verification
+
+For certain sensitive profile data, such as a new password or a verified email that can be used to identify the user, a verification record is required to ensure the ownership and validity of the data. Logto supports the following types of new profile data verification records:
+
+- **New password**: This verification record is used to set up a new password for the user. It checks the new password against the password policy settings and, for existing users, against the old password to prevent reuse. For new identity registrations, a related identifier must be provided to verify the password-based identity availability against the existing user database.
+- **TOTP secret**: This verification record is used to set up a new TOTP MFA method for the user. It verifies the TOTP secret and generates a new TOTP code for the user to verify the TOTP MFA method.
+- **WebAuthn registration**: This verification record is used to set up a new WebAuthn MFA method for the user. It generates a new WebAuthn registration options object for the user to verify the WebAuthn MFA method.
+
+### 4.4 User profile
+
+The user profile consists of the user data that will be stored in the Logto database. Any new profile data provided by the user in an interaction will be saved once the interaction is submitted.
+Developers can use Logto sign-in experience settings to define the required profile data and MFA settings that need to be collected from the user for submitting the interaction.
+
+#### 4.4.1 User identifier
+
+[User identifiers](#42-identifiers) can be configured as a requirement for the profile fulfillment stage. The user must provide the necessary identifiers to identify themselves during the interaction.
```

Comment:
When we are talking about "profile fulfillment stage", does readers know other stages and the order of all stages in a sign-in/sign-up flow? If not, we should better provide a flow chart to show this.

Reply:
Remove the `profile fulfillment stage` word, as technically, readers do need to understand the concept here. 
Also the interaction status was introduces in the following Section 5.